### DISCUSSION TITLE
Navigation options new design

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#58] [Q&A] [answered]
We can redesign the navigation **options** screen with large icon buttons for better usability.
(appears when **long press** the top left navigation panel)

Please let me know if you think the button order should be different.

The 4th option is a placeholder for symmetry, any suggestions?

---

|Rerouting|Auto zoom|
|:-:|:-:|
|**Voice guidance**|**?**|

<img src="https://user-images.githubusercontent.com/3484020/222767125-89e9aa58-797d-4733-b0b5-755cf3addeab.jpg" width="50%"/>
<img src="https://user-images.githubusercontent.com/3484020/222767142-48c58b17-f8a9-4a76-a598-ce8ef0df0232.jpg"/>

**Comment by Leeb64** [comment]
In general, the touch fields in the navigation options could be designed a little better to make them easier to recognize. Maybe with a bold frame or different colors...
Maybe a bit bigger to avoid operating errors - on the motorcycle with gloves.

**Reply by devemux86** [reply]
Indeed, this is the purpose of the new design.
Large buttons with colored icons and make them easier to press with gloves.

**Comment by WolfgangZ42** [comment]
What about "Route type"? We then could switch between "Fastest" and "Shortest" - the travel modes of car and motorcycle has no differences anyway. 
And could you please call it "Recalculation" instead of "Rerouting" ... 🤔

**Reply by devemux86** [reply]
> And could you please call it "Recalculation" instead of "Rerouting"

No problem! :slightly_smiling_face: 
I mentioned the texts of the existing screens.

> What about "Route type"?

The "Route type" is a routing option, it is used in the route calculations.
Is it useful in an active navigation where there is already a calculated route?

**Reply by WolfgangZ42** [reply]
Touché - I didn't think about an active route. Thought of coming into rain and change to "fastest" then.

**Comment by Leeb64** [comment]
A short route is usually suboptimal. Because of the 3m savings, strange shortcuts are taken....

**Reply by WolfgangZ42** [reply]
You are right - but here "shortest" is just an indicator for BRouter to take the right profile.

**Comment by devemux86** [comment]
We are looking for an option that we may change frequently during navigation.
You can look in the Cruiser navigation or other settings to get ideas.

e.g. Strict route following?

**Comment by DolomitesTom** [comment]
WOW button to store the actual GPS position for later review.

**Comment by devemux86** [comment]
In the future we may add settings to customize the navigation buttons (actions / options).
For now we are looking the existing actions / options. :slightly_smiling_face:

**Comment by DolomitesTom** [comment]
Both options sound to be interesting as 4th position. But I don't know how are the effects when suddenly switching off strict follow route in connection with Brouter offline. I have not used any other option since years. What I don't want in any case is driving through bigger cities only because some important shaping points get deleted when I leave the planned route (by default I use automatically recalculate route). And switching routing type would only affect the actual section or the whole route?

**Reply by devemux86** [reply]
> And switching routing type would only affect the actual section or the whole route?

Do you mean the "Settings | Routing | Options | Route type"?
If affects new route calculations or the partial rerouting _until next waypoint_.

**Reply by DolomitesTom** [reply]
Yes, a toggle between short and quick route could be useful, but also a toggle for strict follow route. I think I vote for first option.

**Comment by DolomitesTom** [comment]
Other idea: what about navigate to the nearest fuel station as 4th action button?

**Reply by devemux86** [reply]
> what about navigate to the nearest fuel station as 4th action button?

We could do that, but the feature must be implemented first,
as we would also need some visual representation (like https://github.com/devemux86/cruiser/discussions/9).

Also that would be an **action** (with tap), discussed in https://github.com/devemux86/cruiser/discussions/57.
Here it is about **options** (with long press).

**Comment by DolomitesTom** [comment]
Ok then I vote for routing type or strict navigation.

**Comment by devemux86** [comment]
Screen saver on / off?

**Reply by 0709wiwiwi** [reply]
Screen saver AND power saver on/off. 

Offer also power saving for LCD displays by min backlight operation.

Minimum backlight brightness (= the LCD power saver) and this combined with a large black  (curtain) button that completely covers  the display (= the OLED power and screen saver). 

The large black (curtain) button is necessary this to avoid any uncontrolled touches of the still activ touch panel. 

Serves so power saving for both display techniques LCD & OLED screen saver.

Goal 
Does not need any device admin rights that are necessary to trigger a display off command.

To know.
Minimum backlight brightness is only a little bit less power efficiënt than a completely dimmed display.

Purpose.
This to avoid as such the very annoying PIN unlock request when for example a text edit is than necessary into the Cruiser app.

**Reply by devemux86** [reply]
> Offer also power saving for LCD displays by min backlight operation.

This is already in my implementation list. :slightly_smiling_face: 

_@0709wiwiwi please also write the feature request in new topic, so we can all discuss it there._

**Reply by 0709wiwiwi** [reply]
Done #61

**Comment by devemux86** [comment]
See edited 1st post for preview images (portrait / landscape): https://github.com/devemux86/cruiser/discussions/58#discussion-4894030
Toggle buttons with on/off status, the enabled ones are colored.

_The 4th option is currently a placeholder for symmetry._

(texts are kept to a minimum to avoid overflow with large fonts)

**Comment by devemux86** [comment]
New preview images with the "Route type" (portrait / landscape) in https://github.com/devemux86/cruiser/discussions/58#discussion-4894030.

We could increase the options if necessary in the future.

**Comment by devemux86** [comment, marked as answer]
The new navigation options screen is available in [Cruiser 3.0.11](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
(long press navigation panel)

Please let me know how you see it in both the light and dark app themes.

The 4th option is not final, we can always change it.

**Reply by DolomitesTom** [reply]
it looks great

**Reply by devemux86** [reply]
Note that the navigation options screen contains toggle on / off buttons.
We turn on / off the navigation options and then press OK to save them.

If you find the OK (native Android) button difficult to operate outside,
I could change it and the dialog saves + closes with each on / off press?
(then we can change only 1 option at a time)

**Reply by DolomitesTom** [reply]
yes better confirm any action change with auto-ok and auto-close. It is easier to tap for the menu again.

**Reply by airald** [reply]
I think I agree with @DolomitesTom . Usually only changes one item.

I use dark theme, but both color schemes are visually friendly for me.

**Reply by devemux86** [reply]
Auto close in navigation options is implemented in [Cruiser 3.0.13](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by devemux86** [comment]
> the dialog saves + closes with each on / off press?
(then we can change only 1 option at a time)

It will work like this in the next Cruiser update.

**Comment by DolomitesTom** [comment]
works perfect for skipping one WP/SP, not yet for the skip several SPs button. But I don't if you planned this or not.

**Reply by devemux86** [reply]
> I don't if you planned this or not

Redesign of skip many waypoints screen is already in my list: :slightly_smiling_face: 

- https://github.com/devemux86/cruiser/discussions/17#discussioncomment-5384275
